src/urlschemehandler.cpp

```diff
@@ -181,9 +181,15 @@ UrlSchemeHandler::replyZimNotFoundPage(QWebEngineUrlRequestJob *request,
                                        const QString &zimId)
 {
     QBuffer *buffer = new QBuffer;
-    auto& book = KiwixApp::instance()->getLibrary()->getBookById(zimId);
-    QString path = QString::fromStdString(book.getPath());
-    QString name = QString::fromStdString(book.getName());
+    QString path = "N/A", name = "N/A";
+    try 
+    {
+        auto& book = KiwixApp::instance()->getLibrary()->getBookById(zimId);
+        path = QString::fromStdString(book.getPath());
+        name = QString::fromStdString(book.getName());
+    }
+    catch (...) { /* Blank */ }
+
```

Comment:
I think that this change is unrelated to directory monitoring - it can be merged into the previous commit.

Reply:
Done